src/Models/ArtistsModel.php

```diff
@@ -112,4 +112,21 @@ public function getArtistById(int $artistID): Artist
         $query->execute(['artistID'=> $artistID]);
         return $query->fetch();
     }
+
+    /**
+     * @return Artist[]
+     */
+    public function getFavoritedSongsByArtistId(int $artistID): array
+    {
+        $query = $this->db->prepare("SELECT `artists`.`id`, `songs`.`id`, `songs`.`song_name`, `songs`.`play_count`, `songs`.`length`, `songs`.`favourite`
```

Comment:
I suspect there is a problem hidden here, because you're selecting the artist id and song id, you'll only get the sond id in your results. So the id of the artists returned here will actually be song ids. If you don't actually need the song id I suggest you remove it.

Also, these seems like it's getting songs, not artists. It would make more sense if this were actually in the song model, and if it returned a song entity instead of an arist that isn't actually an artist

Reply:
We needed this or everything breaks and will add a month to the estimate of finish of the project